workers/worker_base.rb

```diff
@@ -25,7 +25,11 @@ def initialize
           # We have the `work` inside a `begin rescue`
           # so that if something fails, the thread still is alive
           begin
-            self.scheduler.schedule { self.work } unless self.should_stop
+            if self.should_stop
+              self.scheduler.shutdown
+            else
+              self.scheduler.schedule { self.work }
```

Comment:
Copying @taquitos comment 
>because this is happening in a loop, does this mean self.scheduler.schedule { self.work } will be called many times? In this case, we might want to consider removing the thread and just using self.scheduler. I think this is the way to move forward because the cron schedule already takes care of repetition, right? So there's no need to call this in a loop.

This was my thought earlier about having a differentiation between cron jobs and workers. What do you think about adding these as types to the worker?

Reply:
I wonder if we can just replace the normal workers the rely on sleep intervals with a scheduler instead? So if somebody wants to run this every 10 seconds, we just create a cron job that repeats every 10 seconds.

Reply:
YES!
![](https://media.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif)

Reply:
done.